<h2>Soal 1: Membuat Objek DataFrames</h2>

- Buatlah objek dataframes dari objek dictionary, Bebas!

In [1]:
# Membuat dataframes dari dictionary

family = {"Name":["Digital","Skola","Data Science"],
            "ID": ["kepala keluarga","istri/ibu","anak"],
            "Birth":["Jakarta, April 6 2021","Bandung, June 22 2020","Surabaya, December 10 2020"],
            "Occupation":["Data Scientist","co-assistant dentist","not working"],
            "Marital Status":["Married","Married","Have not Married"]}

In [2]:
import pandas as pd

In [4]:
df_family = pd.DataFrame(family)
display(df_family)

Name               ID                     Birth  \
0       Ahmad Wali Radhi  kepala keluarga       Medan, March 8 1996   
1               Tsuroyya        istri/ibu  Bojonegoro, July 21 1996   
2  Ishmah Rusyda Haniyya             anak     Jombang, July 10 2020   

             Occupation    Marital Status  
0            Freelancer           Married  
1  co-assistant dentist           Married  
2           not working  Have not Married

<h2>Soal 2: Membaca data dari file</h2>

Download file bernama [diamonds.csv](https://drive.google.com/uc?export=download&id=1qQiM9utQUThTPq8jk8stPMh_y2zfqYOd), kemudian baca data tersebut dan assign ke dalam suatu variabel kemudian print.

In [4]:
# Membaca data dan memasukannya kedalam suatu variabel

diamond_info = pd.read_csv("diamonds-1.csv")
display(diamond_info)

Unnamed: 0  carat        cut color clarity  depth  table  price     x  \
0               1   0.23      Ideal     E     SI2   61.5   55.0    326  3.95   
1               2   0.21    Premium     E     SI1   59.8   61.0    326  3.89   
2               3   0.23       Good     E     VS1   56.9   65.0    327  4.05   
3               4   0.29    Premium     I     VS2   62.4   58.0    334  4.20   
4               5   0.31       Good     J     SI2   63.3   58.0    335  4.34   
...           ...    ...        ...   ...     ...    ...    ...    ...   ...   
53935       53936   0.72      Ideal     D     SI1   60.8   57.0   2757  5.75   
53936       53937   0.72       Good     D     SI1   63.1   55.0   2757  5.69   
53937       53938   0.70  Very Good     D     SI1   62.8   60.0   2757  5.66   
53938       53939   0.86    Premium     H     SI2   61.0   58.0   2757  6.15   
53939       53940   0.75      Ideal     D     SI2   62.2   55.0   2757  5.83   

          y     z  
0      3.98  2.43  
1      3.84  2.31  
2      4.07  2.31  
3      4.23  2.63  
4      4.35  2.75  
...     ...   ...  
53935  5.76  3.50  
53936  5.75  3.61  
53937  5.68  3.56  
53938  6.12  3.74  
53939  5.87  3.64  

[53940 rows x 11 columns]

Expected Result:

![alt text](https://drive.google.com/uc?id=1BvgM5mJEVfj0fllrjkBfQwn0QIFkry_t)

<h2>Soal 3: DataFrame vs Series</h2>

- Apa perbedaan Series dengan list dan dictionary
- Apa perbedaan dataframe dan series
- Buatlah suatu data series

Jawab: 

1. Series adalah suatu objek satu dimensi yang dapat menyimpan berbagai jenis tipe data secara struktur seperti integer, string, dan lain sebagainya. tipe data dari objek series ini harus seragam dan hanya memiliki satu sumbu saja. Penggunaan method dan atribut yang dimiliki pada series juga khusus berbeda dengan tipe data list maupun dictionary. Sedangkan list sederhanya hanya untu menyimpan data dalam satu variabel. Bedanya dengan dictionary tidak ada atribut keys dan values.

2. DataFrame adalah suatu objek 2 dimensi tempat menyimpan data dengan lebih terstruktur. dataframe memiliki 2 index, yaitu index baris dan index columns. Dalam satu column dataframe harus memiliki tipe data yang sama. tapi antar columnnya dataframe bisa memiliki jenis data yang berbeda. Adapun Series adalah suatu objek satu dimensi yang dapat menyimpan berbagai jenis tipe data secara struktur seperti integer, string, dan lain sebagainya. tipe data dari objek series ini harus seragam dan hanya memiliki satu sumbu saja.

In [5]:
presidents = pd.Series(["Soekarno","Soeharto","BJ. Habibie","Abdurrahman Wahid","Megawati","SBY","Jokowi"])

display(presidents)

0             Soekarno
1             Soeharto
2          BJ. Habibie
3    Abdurrahman Wahid
4             Megawati
5                  SBY
6               Jokowi
dtype: object

<h2>Soal 4: Pemeriksaan Data Sederhana</h2>

Di soal no 2 kamu telah membaca data tentang diamonds. pada soal ini cobalah suatu metode dari pandas untuk mengetahui beberapa data pertama dan beberapa data terakhir. kemudian sebutkan nilai dari:
- baris pertama dari column price
- baris terakhir dari column color

tunjukan cara kalian menemukan nilai tersebut di block di bawah ini.

In [11]:
diamond_info['price'].head(1)

0    326
Name: price, dtype: int64

In [17]:
diamond_info.loc[0,'price']

326

In [10]:
diamond_info['color'].tail(1)

53939    D
Name: color, dtype: object

<h2>Soal 5: Deskriptif Statistik</h2>

Jelaskan apa itu deskriptis statistik!

Jawab: Statistika deskriptif merupakan suatu metode yang berhubungan dengan pengumpulan atau penyajian data sampai memberi informasi yang berguna. Dengan menggunakan Statistika deskriptif, berbagai kumpulan data bisa tersaji dengan ringkas dan juga rapi serta mampu memberikan informasi inti dari kumpulan data yang ada.

Informasi yang di dapatkan yang berasal dari statistika deskriptif ini antara lain ukuran pemusatan data, ukuran penyebaran data, dan juga kecenderungan suatu gugus data.

Ada dua macam metode dasar di dalam statistik deskriptif, antara lain numerik dan grafis.

Pendekatan numerik bisa dipakai dalam menghitung nilai statistik dari sekumpulan data.
Sebagai contoh: meandan standar deviasi. Statistik ini akan memberikan informasi mengenai rata-rata serta informasi rinci mengenai distribusi data.

Metode grafis lebih sesuai dibandingkan dengan metode numerik untuk mengidentifikasi pola-pola tertentu dalam data, dilain pihak, pendekatan numerik lebih tepat serta objektif.

Dengan begitu, pendekatan numerik dan juga grafis satu sama lain akan saling melengkapi. Maka dari itu, sangatlah bijaksana jika kita memakai kedua metode tersebut secara bersamaan.
Terdapat tiga karakteristik atau ciri utama dari variabel tunggal, diantaranya yaitu:

1. Distribusi data (distribusi frekuensi)
2. Ukuran pemusatan atau tendensi sentral (Central Tendency)
3. Ukuran penyebaran (Dispersion)

<h2>Soal 6: Practice Deskriptif Statistik</h2>

- Gunakan suatu metode dari pandas untuk mengetahui deskriptif statistik dari suatu data
- Berapa nilai rata-rata dari column price
- Berapa nilai standar deviasi dari column depth
- Berapa nilai maximum dari column carat

In [8]:
diamond_info.describe()

Unnamed: 0         carat         depth         table         price  \
count  53940.000000  53940.000000  53940.000000  53940.000000  53940.000000   
mean   26970.500000      0.797940     61.749405     57.457184   3932.799722   
std    15571.281097      0.474011      1.432621      2.234491   3989.439738   
min        1.000000      0.200000     43.000000     43.000000    326.000000   
25%    13485.750000      0.400000     61.000000     56.000000    950.000000   
50%    26970.500000      0.700000     61.800000     57.000000   2401.000000   
75%    40455.250000      1.040000     62.500000     59.000000   5324.250000   
max    53940.000000      5.010000     79.000000     95.000000  18823.000000   

                  x             y             z  
count  53940.000000  53940.000000  53940.000000  
mean       5.731157      5.734526      3.538734  
std        1.121761      1.142135      0.705699  
min        0.000000      0.000000      0.000000  
25%        4.710000      4.720000      2.910000  
50%        5.700000      5.710000      3.530000  
75%        6.540000      6.540000      4.040000  
max       10.740000     58.900000     31.800000

Jawab: 

nilai rata-rata dari column price adalah 3932.799722

nilai standar deviasi dari column depth adalah 1.432621

nilai maksimum dari column carat adalah 5.01
